src/main/kotlin/script/birthday/BirthdayScript.kt

```diff
@@ -0,0 +1,173 @@
+        val users = getAllUser()
+
+        return users.mapNotNull {
+            val userName = it.id.usernameString
+            val daysUntilMsg = when (val daysUntilBirthday = it.birthdate?.let { it1 -> calculateDaysUntilBirthday(currentDate.toLocalDate(), it1) }) {
+                7 -> "in einer Woche"
+                3 -> "in $daysUntilBirthday Tagen"
+                1 -> "morgen"
+                0 -> "heute"
+                else -> null
+            }
+            daysUntilMsg?.let { "$userName hat $daysUntilMsg Geburtstag" }
+        }
+    }
+
+    private fun List<SelectUsersForTeam>.getNextBirthdaysIn7Days(): List<String> {
+        return mapNotNull {
+            val userName = it.id.usernameString
+            val daysUntilMsg = when (val daysUntilBirthday = it.birthdate?.let { it1 -> calculateDaysUntilBirthday(currentDate.toLocalDate(), it1) }) {
+                7 -> "in einer Woche"
+                3 -> "in $daysUntilBirthday Tagen"
+                1 -> "morgen"
+                0 -> "heute"
+                else -> null
+            }
+            daysUntilMsg?.let { "$userName hat $daysUntilMsg Geburtstag" }
+        }
+    }
+
+    private fun notifyUserForIncomingBirthdays(ctx: EventContext) {
+        val users = getAllUser()
+        users.forEach { user ->
+            val nextBirthdaysIn7Days = user.getBirthdaysForUser()
+
+            val response = if (nextBirthdaysIn7Days.isNotEmpty()) {
+                buildString {
+                    appendLine("Geburtstage:")
+                    nextBirthdaysIn7Days.forEach { it ->
+                        appendLine(it)
+                    }
+                }
+            } else {
+                "Keine anstehenden Geburtstage"
+            }
+            ctx.client().chatPostMessage {
+                it.token(ctx.botToken)
+                it.channel(user.id.id)
+                ctx.logger.info(ctx.requestUserId)
+
+                it.text(response)
+            }
+        }
+    }
+    private fun User.getBirthdaysForUser(): List<String> {
+        val teams = getTeamsForUser(id)
+        val teamMember = buildList {
+            teams.forEach { team ->
+                addAll(getUserInTeam(team.id))
+            }
+        }.distinctBy { it.id }.filterNot { it.id == id }
+        return teamMember.getNextBirthdaysIn7Days()
+    }
+
+    private fun getAllUser(): List<User> = userRepository.selectAll()
+
+    private fun getTeamsForUser(userId: UserId): List<Team> {
+        return teamRepository.getTeamsForUser(userId)
+    }
+
+    private fun getUserInTeam(teamId: TeamId): List<SelectUsersForTeam> {
+        return teamRepository.getUsersForTeam(teamId)
+    }
+
+    private fun getBirthdate(birthdate: LocalDate): String? {
+        val formatter = DateTimeFormatter.ofPattern("MM.dd.yyyy")
+        return birthdate.format(formatter)
+    }
+
+    private fun calculateDaysUntilBirthday(startDate: LocalDate, endDate: LocalDate): Int {
```

Comment:
Wollen wir noch markieren was davon Inclusive und was Exclusive ist? Bspw (wobei nicht kontrolliert):
```suggestion
    private fun calculateDaysUntilBirthday(startDateInclusive: LocalDate, endDateExclusive: LocalDate): Int {
```